### DISCUSSION TITLE
"discovered_devices_and_advertisement_data" doesn't return dictionary

### DISCUSSION THREAD

**Discussion by jcolebaker** [hbldh/bleak#1323] [Q&A] [answered]
According to the docs, the `BleakScanner.discovered_devices_and_advertisement_data` property should return `Dict[str, Tuple[[BLEDevice], [AdvertisementData]]`

I am trying to use it to get the advertisement data after finding a device by name, but I get an error. Here is the code:

```
import asyncio
from bleak import *

DEVICE_NAME = "MyDevice"

async def test_find_device():
    device = await BleakScanner.find_device_by_name(DEVICE_NAME)

    if device == None:
        print("Device not found!")
        return

    print("discovered_devices_and_advertisement_data: ", BleakScanner.discovered_devices_and_advertisement_data)

    (_, ble_advert_data) = BleakScanner.discovered_devices_and_advertisement_data[DEVICE_NAME]
    print(ble_advert_data)


if __name__ == '__main__':
    loop = asyncio.get_event_loop()
    loop.run_until_complete(test_find_device())
```

I get the following output (Windows 10): 

```
discovered_devices_and_advertisement_data:  <property object at 0x0000012D5CA79310>
Traceback (most recent call last):
  File "C:\Users\Jeremycb\esp\wireless-controller\system_tests\test2.py", line 22, in <module>
    loop.run_until_complete(test_find_device())
  File "C:\Users\Jeremycb\AppData\Local\Programs\Python\Python39\lib\asyncio\base_events.py", line 642, in run_until_complete
    return future.result()
  File "C:\Users\Jeremycb\esp\wireless-controller\system_tests\test2.py", line 15, in test_find_device
    (_, ble_advert_data) = BleakScanner.discovered_devices_and_advertisement_data[DEVICE_NAME]
TypeError: 'property' object is not subscriptable
```

It definitely finds my BLE device, and I can use the returned "device" object to create a client and connect. 

I also tried `BleakScanner.discovered_devices_and_advertisement_data.keys()` and `BleakScanner.discovered_devices_and_advertisement_data.value()` but these don't exist. (`Exception: 'property' object has no attribute 'keys'`, etc).

**Am I using this property incorrectly? Have I made a stupid Python error?** 

```
Python 3.9.2
bleak                     0.20.0
bleak-winrt               1.2.0
```

**Comment by dlech** [comment, marked as answer]
`discovered_devices_and_advertisement_data` is an instance property, so it can't be used this way since you aren't creating an instance of `BleakScanner`. 

Please give https://github.com/hbldh/bleak/issues/1277 a :+1: to show your support for this use case.
